lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -828,13 +813,16 @@ Value TypeLoweringVisitor::addArg(FModuleOp module, Type type,
   auto newValue = body->addArgument(type);
 
   // Save the name attribute for the new argument.
-  StringAttr nameAttr = getFIRRTLNameAttr(module.getArgAttrs(oldArgNumber));
-  if (nameAttr) {
-    auto newArgAttrs = getArgAttrs(nameAttr, nameSuffix, builder);
-    auto newArgNumber = newValue.getArgNumber() - originalNumModuleArgs;
-    auto newArgAttrName = getArgAttrName(newArgNumber);
-    newModuleAttrs.push_back(NamedAttribute(newArgAttrName, newArgAttrs));
-  }
+  StringAttr nameAttr =
+      getFIRRTLModuleArgNameAttr(module)[oldArgNumber].cast<StringAttr>();
+  Attribute newArg;
+  if (!nameAttr.getValue().empty()) {
+    SmallString<16> newName(nameAttr.getValue());
+    newName += nameSuffix;
+    newArg = builder->getStringAttr(newName);
```

Comment:
You can rely on some `llvm::Twine` magic to avoid the string concatenation until `getStringAttr` actually needs it:
```suggestion
    newArg = builder->getStringAttr(nameAttr.getValue() + nameSuffix);
```

Reply:
Thanks Schuyler, made the changes